gantry/utils/collect.py

```diff
@@ -0,0 +1,281 @@
+        }
+    )
+
+    job_requests_res = await prometheus.query(
+        type="single",
+        query={
+            "metric": "kube_pod_container_resource_requests",
+            "filters": {"container": "build", "pod": job["pod"]},
+        },
+        time=query_time,
+    )
+
+    job_limits_res = await prometheus.query(
+        type="single",
+        query={
+            "metric": "kube_pod_container_resource_limits",
+            "filters": {"container": "build", "pod": job["pod"]},
+        },
+        time=query_time,
+    )
+
+    mem_usage = process_usage(
+        await prometheus.query(
+            type="range",
+            query={
+                "metric": "container_memory_working_set_bytes",
+                "filters": {"container": "build", "pod": job["pod"]},
+            },
+            start=job["start"],
+            end=job["end"],
+        ),
+        job["build_id"],
+    )
+
+    cpu_usage = process_usage(
+        await prometheus.query(
+            type="range",
+            custom_query=f"rate(container_cpu_usage_seconds_total{{pod='{job['pod']}', container='build'}}[90s])",
+            start=job["start"],
+            end=job["end"],
+        ),
+        job["build_id"],
+    )
+
+    # instead of needing to fetch the node where the pod ran from kube_pod_info
+    # we can grab it from kube_pod_container_resource_limits
+    # weirdly, it's not available in kube_pod_labels or annotations
+    # https://github.com/kubernetes/kube-state-metrics/issues/1148
+    vm = await fetch_vm(job_limits_res[0]["labels"]["node"], query_time)
+    requests = process_resources_res(job_requests_res)
+    limits = process_resources_res(job_limits_res)
+
+    # TODO insert into db here
+
+    return db.insert(
+        "builds",
+        (
+            None,
+            job["pod"],
+            vm,
+            job["start"],
+            job["end"],
+            job["build_id"],
+            job["build_status"],
+            job["ref"],
+            job["pkg_name"],
+            job["pkg_version"],
+            # dict to string
+            json.dumps(job["pkg_variants"]),
+            job["compiler_name"],
+            job["compiler_version"],
+            job["arch"],
+            job["stack"],
+            job["build_jobs"],
+            requests["cpu"]["value"],
+            # currently not set as of 12-23
+            limits.get("cpu", {}).get("value"),
+            cpu_usage["mean"],
+            cpu_usage["median"],
+            cpu_usage["max"],
+            cpu_usage["min"],
+            cpu_usage["stddev"],
+            requests["memory"]["value"],
+            limits["memory"]["value"],
+            mem_usage["mean"],
+            mem_usage["median"],
+            mem_usage["max"],
+            mem_usage["min"],
+            mem_usage["stddev"],
+        ),
+    )
+
+
+async def fetch_vm(hostname: str, query_time: float) -> dict:
+    prometheus = PrometheusClient()
+    db = SqliteClient()
```

Comment:
an `SqliteClient` was created above in `fetch_job`...what's the best way to share the client across the application?